Scripts/Runtime/Entities/TaskSystem/AbstractTaskSystem.cs

```diff
@@ -27,99 +27,126 @@ public abstract class AbstractTaskSystem : AbstractAnvilSystemBase
 
         private BulkJobScheduler<TaskDriverCancellationPropagator> m_TaskDriversCancellationBulkJobScheduler;
 
+        private TaskFlowGraph m_TaskFlowGraph;
         private bool m_IsHardened;
-        
+
         /// <summary>
         /// The context of this <see cref="AbstractTaskSystem"/>.
         /// This will always be the first ID given by a <see cref="ByteIDProvider"/> and is used to differentiate
         /// between instances running in the generic system job(s) versus more specific Task Drivers.
         /// </summary>
         public byte Context { get; }
 
+        internal CancelRequestsDataStream CancelRequestsDataStream { get; }
+        internal List<AbstractTaskStream> TaskStreams { get; }
+        internal List<AbstractTaskDriver> TaskDrivers { get; }
+
+
         protected AbstractTaskSystem()
         {
+            //TODO: #112 (anvil-csharp-core) Extract to Anvil-CSharp Util method -Used in AbstractJobConfig as well
+            m_TypeString = GetType().Name;
+
             m_TaskDriverContextProvider = new ByteIDProvider();
-            m_TaskDrivers = new List<AbstractTaskDriver>();
+            TaskStreams = new List<AbstractTaskStream>();
+            TaskDrivers = new List<AbstractTaskDriver>();
+            m_JobConfigs = new List<AbstractJobConfig>();
 
             Context = m_TaskDriverContextProvider.GetNextID();
+
+            TaskStreamFactory.CreateTaskStreams(this, TaskStreams);
+            CancelRequestsDataStream = new CancelRequestsDataStream();
+        }
+
+        protected override void OnCreate()
+        {
+            base.OnCreate();
+            //Initialize the TaskFlowGraph based on our World
+            InitTaskFlowGraph(World);
+        }
+
+        private void InitTaskFlowGraph(World world)
+        {
+            //We could get called multiple times 
+            if (m_TaskFlowGraph != null)
+            {
+                return;
+            }
             
-            //TODO: #71 - Let the TaskFlowGraph handle creating this for us.
-            m_CancelRequestsDataStream = new CancelRequestsDataStream();
-
-            //TODO: #65 - Need to look at having this happen in OnCreate instead. The World is only set there. 
-            World currentWorld = World ?? World.DefaultGameObjectInjectionWorld;
-            m_TaskFlowGraph = currentWorld.GetOrCreateSystem<TaskFlowSystem>().TaskFlowGraph;
-            m_TaskFlowGraph.CreateTaskStreams(this);
-            m_TaskFlowGraph.RegisterCancelRequestsDataStream(m_CancelRequestsDataStream, this, null);
+            m_TaskFlowGraph = world.GetOrCreateSystem<TaskFlowSystem>().TaskFlowGraph;
+            m_TaskFlowGraph.RegisterTaskSystem(this);
```

Comment:
🚲 May be worth just having a `Register` method on TaskFlow graph with overloads for each type it can handle?

Reply:
Added TODO for #66, #67, and/or #68 and referenced this comment.